dali/operators/reader/nvdecoder/nvdecoder.cc

```diff
@@ -57,6 +58,37 @@ NvDecoder::NvDecoder(int device_id,
     "Failed to load libcuda.so. "
     "Check your library paths and if NVIDIA driver is installed correctly.");
 
+
+  // This is a workaround for an issue with nvcuvid in drivers >460 where concurrent
+  // use on default context and non-default streams may lead to memory corruption.
+  // TODO(michalz): add an upper bound when the problem is fixed
+  bool use_default_stream = false;
+#if NVML_ENABLED
+  {
+    static float driver_version = []()->float {
+      char version[80];
+      CUDA_CALL(nvmlInitChecked());
+      CUDA_CALL(nvmlSystemGetDriverVersion(version, sizeof version));
+
+      return std::stof(version);
+    }();
+    if (driver_version > 460)
+      use_default_stream = true;
+  }
+#else
+  {
+    int driver_cuda_version = 0;
+    CUDA_CALL(cuDriverGetVersion(&driver_cuda_version));
```

Comment:
Why can't use this in all cases?

Reply:
Maybe we'll be able to cut it off somewhere between versions that differ in supported CUDA version. If it coincides, we can drop NVML part.